jmclient/test/test_yieldgenerator.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import (absolute_import, division,
+                        print_function, unicode_literals)
+from builtins import * # noqa: F401
+
+import unittest
+
+from jmclient import jm_single, YieldGeneratorBasic
+
+
+class FakeWallet:
```

Comment:
Just to mention, we already have DummyWallet as a test class.

Reply:
Yes, I'd prefer using the existing `DummyWallet`, too. Implementing dedicated mock wallets for each test will just cause considerable overhead when changing something.

Reply:
I know, I've seen the `DummyWallet`.  However, it seemed to have "just" a bunch of hardcoded UTXOs in it, and I didn't see an easy way to a) set the balances of each mixdepth from the test, and b) e.g. verify that the UTXOs or addresses returned are from a certain mixdepth.  Maybe I just missed how to do it.

In other words, it seemed to me that `DummyWallet` was much more geared towards testing lower-level taker/maker stuff, rather than the abstract level of the yield generator.

Reply:
Indeed, with the `DummyWallet` from `test_taker` it's actually hard to tell how it's going to behave outside of that test suite. Is anything wrong with using the actual wallet implementation?

Reply:
I've taken a brief look at the actual wallet, and it seems like we could make it do all things required for the test relatively easily.  I'll try that out later and will switch to it if it seems possible.

Reply:
The test is now updated to use a real wallet (with some tiny utility functions added on top).